compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -304,6 +303,13 @@ private void defineConstructs(BLangPackage pkgNode, SymbolEnv pkgEnv) {
         pkgNode.typeDefinitions.forEach(typDef -> typDefs.add(typDef));
         defineTypeNodes(typDefs, pkgEnv);
 
+        for (BLangSimpleVariable variable : pkgNode.globalVars) {
+            if (variable.expr != null
+                    && variable.isDeclaredWithVar && variable.expr.getKind() == NodeKind.LAMBDA) {
+                addFunctionType(variable, pkgEnv);
```

Comment:
Shall we rename `addFunctionType` to something descriptive.
suggestion: resolveAndSetFunctionTypeFromRHSLambda

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/19976/commits/7432217a84fd35c7ae7682d1e9173e9416f9ccdc